CSharp/Library/QnAMaker/QnAMakerDialog.cs

```diff
@@ -1,110 +1,225 @@
+                await this.DefaultWaitNextMessageAsync(context, message, qnaMakerResults);
+            }
+        }
+
+        protected virtual bool IsConfidentAnswer(QnAMakerResults qnaMakerResults)
+        {
+            if (qnaMakerResults.Answers.Count < 2 || qnaMakerResults.Answers.FirstOrDefault().Score >= 99)
+            {
+                return true;
+            }
+
+            if (qnaMakerResults.Answers[0].Score - qnaMakerResults.Answers[1].Score > 20.0)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private async Task ResumeAndPostAnswer(IDialogContext context, IAwaitable<string> argument)
+        {
+            var selection = await argument;
+
+            var qnaMakerResults = default(QnAMakerResults);
+            context.PrivateConversationData.TryGetValue("qnaMakerResults", out qnaMakerResults);
+
+            if (qnaMakerResults != null)
+            {
+                bool match = false;
+                foreach (var qnaMakerResult in qnaMakerResults.Answers)
+                {
+                    if (qnaMakerResult.Questions[0].Equals(selection, StringComparison.OrdinalIgnoreCase))
+                    {
+                        context.PostAsync(qnaMakerResult.Answer);
+                        match = true;
+
+                        var feedbackRecord = default(FeedbackRecord);
+                        context.PrivateConversationData.TryGetValue("feedbackRecord", out feedbackRecord);
+
+                        if (feedbackRecord != null)
+                        {
+                            feedbackRecord.KbQuestion = qnaMakerResult.Questions.FirstOrDefault();
+                            feedbackRecord.KbAnswer = qnaMakerResult.Answer;
+
+                            var tasks =
+                                this.services.Select(
+                                    s =>
+                                    s.ActiveLearnAsync(
+                                        feedbackRecord.UserId,
+                                        feedbackRecord.UserQuestion,
+                                        feedbackRecord.KbQuestion,
+                                        feedbackRecord.KbAnswer)).ToArray();
+                            break;
+                        }
+                    }
+                }
+                if (!match)
+                {
+                    context.PostAsync(
+                        "Not able to match. Please click on the options or type in the exact text from the options.");
+                }
+            }
+            await this.DefaultWaitNextMessageAsync(context, context.Activity.AsMessageActivity(), qnaMakerResults);
+        }
+
+        protected virtual async Task QnAFeedbackStepAsync(IDialogContext context, QnAMakerResults qnaMakerResults)
+        {
+            var qnaList = qnaMakerResults.Answers;
+            var questions = qnaList.Select(x => HttpUtility.HtmlDecode(x.Questions[0])).ToArray();

```

Comment:
We don't allow empty question in QnA Maker for any Answer. So it can't be empty.
Also, every answer can have more than 1 questions mapped to it using the "Add alternative phrase" options in QnA Maker tool. For the dialog, we want to show only one question for each answer.